vir/defs/polymorphic/ast/common.rs

```diff
@@ -171,6 +171,16 @@ pub enum Type {
     TypeVar(TypeVar),
 }
 
+impl Type {
+    pub fn map(key_type: Type, val_type: Type) -> Type {
```

Comment:
I added the `map` function there to construct the domain type of the map, as I didn't see it suitable in other parts of the code.
I have since been informed that in newer versions of viper there is a builtin map type, which removes the need for the domain and this function.